website/source/docs/guides/connect-gateways.md

```diff
@@ -0,0 +1,348 @@
+across multiple datacenters. Now you are ready to register the services that
+will use Connect.
+
+## Register a Service in Each Datacenter to Use Connect
+
+You can register a service to use a sidecar proxy by including a sidecar proxy
+stanza in its registration file. For this guide, you can use netcat to act as a
+backend service and register a dummy service called web to represent the client
+service. Those names are used in our examples. If you have services that you
+would like to connect, feel free to use those instead.
+
+~> **Caution:** Connect takes its default intention policy from Consul’s default
+ACL policy. If you have set your default ACL policy to deny (as is recommended
+for secure operation)  and are adding Connect to already registered services,
+those services will lose connection to each other until you set an intention
+between them to allow communication.
+
+### Register a back end service in one datacenter
+
+In one datacenter register a backend service and add an Envoy sidecar proxy
+registration. To do this you will either create a new registration file or edit
+an existing one to include a sidecar proxy stanza. If you are using netcat as
+your backend service, you will create a new file called `netcat.json` that would
+contain the below snippet. Since you have ACLs enabled, you will have to create
+a token for the service.
+
+```
+{
+  "service": {
+    "name": "netcat",
+    "port": 8181,
+    "token": <token here>,
+    "connect": {"sidecar_service": {} }
+
+}
+```
+
+Note the Connect part of the registration with the sidecar service and token
+options. This is what you would add to an existing service registration if you
+are not using netcat as an example.
+
+Reload the client with the new or modified registration.
+
+```
+consul reload
+```
+
+Then start Envoy specifying which service it will proxy.
+
+```
+consul connect envoy -sidecar-for netcat
+```
+
+If you are using netcat as your example, start it now by running the following
+command. netcat will run in your terminal window and echo back anything you type
+on the command line.
+
+```
+$ nc 127.0.0.1 8181
+hello
+hello
+echo
+echo
+```
+
+Stop the running service by typing `ctrl + c`.
+
+### Register a front end service in the other datacenter
+
+Now in your other datacenter, you will register a service that calls your
+backend service, with a sidecar proxy Your registration will need to list the
+backend service as your upstream. Like the backend service, you can use our
+example service, which will be called web, or append the connect stanza to an
+existing registration with some customization.
+
+To use web as your front end service, create a registration file called
+`web.json` that contains the following snippet.
+
+```
+{
+  "service": {
+    "name": "web",
+    "port": 8080,
+    "token": <token here>,
+    "connect": {
+      "sidecar_service": {
+        "proxy": {
+          "upstreams": [{
+             "destination_name": "netcat",
```

Comment:
The upstream config also needs a datacenter argument or else they will look in the local datacenter for the service:

```
upstreams = [{
  destination_name = "netcat"
  datacenter = "primary"
  local_bind_port = 8181
}]
```